### PR TITLE
Fix #151 and another bug I found along the way

### LINKED ISSUE #151: Do not complain about repo files when no packages from the repo are installed.
We want to change the blocker to block if rpms are installed from a repo we don't know about. 

We want to warn if repo files are present and not fully disabled when unknown.

### PR DESCRIPTION
**Pull request by troglodyne**
By submitting pull requests to this repo, I agree to the Contributor License Agreement which can be found at: https://github.com/cpanel/elevate/blob/main/docs/cPanel-CLA.pdf

See second commit in set for "the bug I found along the way."

Internal tracking/testing notes can be found in BC-6268.

### CODE REVIEW

File: elevate-cpanel [hunk 1]
```diff
@@ -232,43 +232,134 @@ use constant GRUB_EDITENV      => '/usr/bin/grub2-editenv';
...
+  extras
+  centosplus
+  cr
+  base-debuginfo
+  c7-media
+  base-source
+  updates-source
+  extras-source
+  centosplus-source
+  C7.0.1406-base
```
Comment (toddr) [inline elevate-cpanel:291]: Is this for cloudlinux? we don't support that yet.

Comment (troglodyne) [reply elevate-cpanel:291]: That was just names of stuff I pulled from the centos-vault repo. No idea why those all looked so odd.

File: elevate-cpanel [hunk 2]
```diff
@@ -232,43 +232,134 @@ use constant GRUB_EDITENV      => '/usr/bin/grub2-editenv';
...
-
+  mysql-connectors-community
+  mysql-tools-community
+  mysql55-community
+  mysql56-community
+  mysql57-community
+  mysql80-community
+  mysql-tools-preview
+  mysql-cluster-7.5-community
+  mysql-cluster-7.6-community
```
Comment (atoomic) [inline elevate-cpanel:243]: maybe we should consider RegExp for some of these entries? which would provide an easier maintenance
example: `mysql-cluster-\d+\.\d+-community`

File: elevate-cpanel [hunk 3]
```diff
@@ -232,43 +232,134 @@ use constant GRUB_EDITENV      => '/usr/bin/grub2-editenv';
...
+  C7.7.1908-base
+  C7.7.1908-updates
+  C7.7.1908-extras
+  C7.7.1908-centosplus
+  C7.7.1908-fasttrack
+  C7.8.2003-base
+  C7.8.2003-updates
+  C7.8.2003-extras
+  C7.8.2003-centosplus
+  C7.8.2003-fasttrack
```
Comment (atoomic) [inline elevate-cpanel:335]: same here RegExp++? (apply to other names where it makes sense too)

File: elevate-cpanel [hunk 4]
```diff
@@ -2265,7 +2356,6 @@ sub yum_list ( $invalidate_cache = 0 ) {
     while ( my $line = shift @lines ) {
         last if $line && $line =~ m/^Installed Packages/;
     }
-    shift @lines;
```
Comment (atoomic) [inline elevate-cpanel:2268]: indeed not sure why that `shift` was there... the first line after `Installed Packages` seems a legit package

File: elevate-cpanel [hunk 5]
```diff
@@ -232,43 +232,134 @@ use constant GRUB_EDITENV      => '/usr/bin/grub2-editenv';
...
-  kernelcare.repo
-  wp-toolkit-cpanel.repo
-  imunify360-ea-php-hardened.repo
-  droplet-agent.repo
+  base
+  updates
+  extras
+  centosplus
+  cr
+  base-debuginfo
```
Comment (toddr) [inline elevate-cpanel:285]: no debuginfo

File: elevate-cpanel (hunk 2 above)
Comment (toddr) [reply elevate-cpanel:243]: Let's do this in a follow up commit

**Pull request merged**